lib/cocoapods/command/repo/push.rb

```diff
@@ -231,7 +231,7 @@ def add_specs_to_repo
             end
 
             # only commit if modified
-            if repo_git('status', '--porcelain').include?(spec.name)
+            if repo_git('status', '--untracked-files=all', '--porcelain').include?(spec.name)
```

Comment:
perhaps we should pass in `spec.name` to the `git status` invocation so we only check that single file, rather than potentially looking at a large amount of untracked files